appliance/mariadb/cmd/flynn-mariadb-api/main.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	_ "github.com/flynn/flynn/Godeps/_workspace/src/github.com/go-sql-driver/mysql"
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/julienschmidt/httprouter"
+	"github.com/flynn/flynn/appliance/mariadb"
+	"github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flynn/pkg/httphelper"
+	"github.com/flynn/flynn/pkg/random"
+	"github.com/flynn/flynn/pkg/resource"
+	"github.com/flynn/flynn/pkg/shutdown"
+)
+
+var serviceName = os.Getenv("FLYNN_MYSQL")
+var serviceHost string
+
+func init() {
+	if serviceName == "" {
+		serviceName = "mariadb"
+	}
+	serviceHost = fmt.Sprintf("leader.%s.discoverd", serviceName)
+}
+
+func main() {
+	defer shutdown.Exit()
+
+	dsn := &mariadb.DSN{
+		Host:     serviceHost + ":3306",
+		User:     "flynn",
+		Password: os.Getenv("MYSQL_PWD"),
+		Database: "mysql",
+	}
+	db, err := sql.Open("mysql", dsn.String())
+	api := &API{db}
+
+	router := httprouter.New()
+	router.POST("/databases", api.createDatabase)
+	router.DELETE("/databases", api.dropDatabase)
+	router.GET("/ping", api.ping)
+
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "3000"
+	}
+	addr := ":" + port
+
+	hb, err := discoverd.AddServiceAndRegister(serviceName+"-api", addr)
+	if err != nil {
+		shutdown.Fatal(err)
+	}
+	shutdown.BeforeExit(func() { hb.Close() })
+
+	handler := httphelper.ContextInjector(serviceName+"-api", httphelper.NewRequestLogger(router))
+	shutdown.Fatal(http.ListenAndServe(addr, handler))
+}
+
+type API struct {
+	db *sql.DB
+}
+
+func (a *API) createDatabase(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
```

Comment:
These handlers should take a context as the first argument. There are examples in a few other APIs including the router: https://github.com/flynn/flynn/blob/master/router/api.go